src/lib.rs

```diff
@@ -27,16 +27,28 @@ use std::time::Duration;
 pub const DEFAULT_GAS_LIMIT: f64 = 21000.0;
 pub const DEFAULT_TIME_LIMIT: Duration = Duration::from_secs(30);
 
+#[derive(Debug, Default, PartialEq)]
+pub struct GasPrice1559 {
+    max_fee_per_gas: f64,
+    max_priority_fee_per_gas: f64,
+}
+
+#[derive(Debug, Default, PartialEq)]
+pub struct GasPrice {
+    gas_price: f64,
+    gas_price_1559: Option<GasPrice1559>,
```

Comment:
uber-naming-nit: I find the `gas_price` name a bit repetiive given that this is a `GasPrice` estimate. I suggest:
```suggestion
    legacy: f64,
    eip1559: Option<GasPrice1559>,
```

Reply:
I find your naming better. Fixed.